vapour_api/api.py

```diff
@@ -0,0 +1,187 @@
+from flask_restful import Resource, Api
+from flask import send_from_directory
+from vapour_api import app
+from geojson import Feature, GeometryCollection, Point
+from flask_sqlalchemy import SQLAlchemy
+from flask_cors import CORS, cross_origin
+from collections import OrderedDict
+import psycopg2
+import ast
+import os
+
+db = SQLAlchemy(app)
+CORS(app)
+app.config["CORS_HEADERS"] = "Content-Type"
+
+ATTRIBUTION = "Jungle Bus, <a href='https://www.openstreetmap.org/copyright/'>© OpenStreetMap contributors</a>"
+
+app.config["SQLALCHEMY_DATABASE_URI"] = "postgresql://{}:{}@{}:{}/{}".format(
+    os.getenv("POSTGRES_USER", ""),
+    os.getenv("POSTGRES_PASSWORD", ""),
+    os.getenv("POSTGRES_HOST", ""),
+    os.getenv("POSTGRES_PORT", ""),
+    os.getenv("POSTGRES_DB", ""),
+)
+
+
+def get_route_properties(route_id):
+    fields = [
+        "osm_id",
+        "name",
+        "ref",
+        "network",
+        "origin",
+        "destination",
+        "colour",
+        "operator",
+    ]
+    result = db.engine.execute(
+        "SELECT {} FROM i_routes where osm_id = {}".format(", ".join(fields), route_id)
+    )
+    row = result.next()
+    route = {}
+    for i, f in enumerate(fields):
+        route[f] = row[i]
+    return route
+
+
+def get_route_geojson(route_id):
+    sql = """
+        SELECT ST_asgeojson(ST_MakeLine((ST_Transform(geom, 4326))))
+        FROM i_ways
+        WHERE rel_osm_id = {}
+        order by index
+    """
+    result = db.engine.execute(sql.format(route_id))
+    row = result.next()
+    return row[0]
+
+
+def get_route_stop_positions(route_id):
+    result = db.engine.execute(
+        """
+            SELECT 
+                pos,
+                st_asGeoJSON(ST_Transform(geom, 4326)) as geojson, 
+                st_X(ST_Transform(geom, 4326)) as lon,
+                st_Y(ST_Transform(geom, 4326)) as lat
+            FROM (
+                SELECT rel_osm_id, unnest(positions_ids) as pos
+                FROM d_routes_position_ids
+            ) t 
+            INNER JOIN d_routes_position 
+                on t.pos = d_routes_position.id
+            WHERE rel_osm_id = {} 
+            ORDER BY pos
+        """.format(
```

Comment:
`str.format()` should not be used with SQL string. Use in place `execute()` arguments to avoid SQL injections
```
engine.execute("insert into users values (?, ?)", 1, "john")
```

Reply:
you're right, thanks !